daemon/lib/lightning/helpers.go

```diff
@@ -0,0 +1,46 @@
+package lightning
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+// ParsePubKey parses a hex-encoded public key (bitcon secp256k1) string into a btcec public key object
+func ParsePubKey(pubKeyStr string) (*btcec.PublicKey, error) {
+	pubKeyBytes, err := hex.DecodeString(pubKeyStr)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey, err := btcec.ParsePubKey(pubKeyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return pubKey, nil
+}
+
+// ValidatePreimage validates that a preimage matches a payment hash
+func ValidatePreimage(preimage string, paymentHash string) error {
```

Comment:
I do find this pkg really good https://pkg.go.dev/github.com/lightningnetwork/lnd/lntypes